frontend/leads-search/lead-list/lead-search-dsl.helpers.tsx

```diff
@@ -0,0 +1,101 @@
+import queryString from "query-string";
```

Comment:
Instead of copy/pasting this entire file and test, let's reuse the client-search-dsl.helpers.tsx.

The functions inside of that file are almost fully reusable, with the only thing needing to change being the allowedSearchFields. Let's allow the allowedSearchFields to be passed in as an argument to the `parseSearch` and `serializeSearch`.

The reason I think that this one is worth consolidating into one file is that it's not a concrete react component that would get more complex as it has to handle more things. It's a file that is already made to be reused, but just needs slight adjustments for the reuse to happen.